test/suites/dev/test-precompile/test-precompile-assets-erc20-local-assets-removal.ts

```diff
@@ -0,0 +1,122 @@
+      "0xfffffffeb1de27b22dcb3a2e1def6d0edefa3bff",
+      "0xfffffffe0b00feb2a14ad90d9d4d871eba59dd9c",
+      "0xfffffffe47b78475160da680caef70959e027bee",
+      "0xfffffffeff0e8e9e2288fcce5e70afa61baec79d",
+      "0xfffffffe4b980819e301da7d7b08b1a64299c5ff",
+      "0xfffffffef0608962d5801b39f437d4cc416939e1",
+      "0xfffffffefd9d0bf45a2947a519a741c4b9e99eb6",
+      "0xfffffffec002b88d59aa53e9612a00fb2a2d1504",
+      "0xfffffffe028682475e21a368c4f6e368efe598f0",
+      "0xfffffffe6d1492e39f1674f65a6f600b4589abd7",
+    ];
+    const notLocalAssets = [
+      "0xfffffffe88fc55121922ed6780b7e0c8ab0374d3",
+      "0xfffffffe221e31a529b1a5fc4101352398974a07",
+    ];
+    let erc20Abi: Abi;
+    let precompileRegistryAbi;
+
+    beforeAll(async () => {
+      const { abi } = await deployCreateCompiledContract(context, "ERC20Instance");
+      erc20Abi = abi;
+      const { abi: _precompileRegistryAbi } = await deployCreateCompiledContract(
+        context,
+        "PrecompileRegistry"
+      );
+      precompileRegistryAbi = _precompileRegistryAbi;
+    });
+
+    it({
+      id: "T01",
+      title: "ensure evm calls on local xc-20 addresses get reverted",
+      test: async function () {
+        const addresses = [
+          ...localAssets.map((address) => ({ address, local: true })),
+          ...notLocalAssets.map((address) => ({ address, local: false })),
+        ];
+
+        // Call every address and ensure it only local asset addresses get reverted
+        for (const { address, local } of addresses) {
+          const rawSigned = await createEthersTransaction(context, {
+            to: address,
+            data: encodeFunctionData({
+              abi: erc20Abi,
+              functionName: "approve",
+              args: [ALITH_ADDRESS, 1000],
+            }),
+          });
+
+          const { result } = await context.createBlock(rawSigned);
+          const receipt = await context
+            .viem("public")
+            .getTransactionReceipt({ hash: result?.hash as `0x${string}` });
+
+          if (local) {
+            // The transaction status should be "reverted" when calling a local asset
+            expect(receipt.status).to.equal("reverted");
+
+            // Assert expected revert reason
+            const revertReason = await extractRevertReason(context, result.hash);
+            expect(revertReason).to.equal("Removed precompile");
+          } else {
+            // The transaction status should be "success"
+            expect(receipt.status).to.equal("success");
+          }
+        }
+      },
+    });
+
+    it({
+      id: "T02",
+      title: "Local assets should be inactive in the precompile registry",
+      test: async function () {
+        const addresses = [
+          ...localAssets.map((address) => ({ address, local: true })),
+          ...notLocalAssets.map((address) => ({ address, local: false })),
+        ];
+
+        for (const { address, local } of addresses) {
```

Comment:
Doing tests in this style can hide true negatives, expect will throw on the first failure.

If you are wanting to verify a list of test data is it much better to instead populate an array of failures and then at the end verify the length of failuresArray is zero.